Fix: use configurationChanges field on deploy

yaronya opened the pull request:
Fixes #49

### Issue & Steps to Reproduce
Align the CLI with how the UI works on initial deploy + redeploy.

### Solution
1. Use `deployRequest` field on initial deloy.
2. Use `configurationChanges` to set proper `DEPLOYMENT` configuration scope on both initial deploy and redeploy.

avnerenv0 commented on node/src/commands/deploy.js:
```diff
@@ -2,41 +2,41 @@ const DeployUtils = require('../lib/deploy-utils');
 
 const assertBlueprintExistsOnInitialDeployment = options => {
   if (!options[BLUEPRINT_ID]) throw new Error('Missing blueprint ID on initial deployment');
+  if (!options[REVISION]) throw new Error('Missing revision on initial deployment');
```
Why?
We don't have to have a revision, every git repo has a default branch.

yaronya replied:
I looked only on `create` where it gets it from request but then on `deploy` it's not required 🤦 
64f9e72

avnerenv0 commented on node/src/commands/deploy.js:
```diff
@@ -2,41 +2,41 @@ const DeployUtils = require('../lib/deploy-utils');
     assertBlueprintExistsOnInitialDeployment(options);
-    environment = await deployUtils.createEnvironment(environmentName, organizationId, projectId);
+    environment = await deployUtils.createAndDeployEnvironment(options, configurationChanges);
+    deployment = await deployUtils.getDeployment(environment.latestDeploymentLogId);
```
this is not needed, we are only using the `id` anyway

yaronya replied:
d7cec36

avnerenv0 commented on node/src/lib/genetal-utils.js:
```diff
@@ -1,8 +1,17 @@
+const removeEmptyValuesFromObj = payload =>
+  JSON.parse(
+    JSON.stringify(payload, (_, value) => {
+      if (value === undefined) return undefined;
```
this function always returns `value` :)
You can `JSON.parse(JSON.stringify(payload));`

avnerenv0 reviewed (approved):
@yaronya left you some comments,
except the `revision` thing, nothing is important 

Makes me understand that any configuration changes given to the deployment API should be scoped as 'DEPLOYMENT' by the backend anyway, the clients don't need to know about this.

yaronya commented:
> Makes me understand that any configuration changes given to the deployment API should be scoped as 'DEPLOYMENT' by the backend anyway, the clients don't need to know about this.

@avnerenv0 TBH It's already scoped by the backend but yet I added it to the request. I sense that this might be confusing. I've removed it from request: 0e5aa01
